src/main/java/teammates/ui/request/FeedbackResponseStatisticCreateRequest.java

```diff
@@ -0,0 +1,207 @@
+
+    public FeedbackParticipantType getRecipientType() {
+        return recipientType;
+    }
+
+    /**
+     * Get number of entities to give feedback to.
+     */
+    public int getNumberOfEntitiesToGiveFeedbackTo() {
+        switch (numberOfEntitiesToGiveFeedbackToSetting) {
+        case CUSTOM:
+            return customNumberOfEntitiesToGiveFeedbackTo;
+        case UNLIMITED:
+            return Const.MAX_POSSIBLE_RECIPIENTS;
+        default:
+            assert false : "Unknown numberOfEntitiesToGiveFeedbackToSetting: " + numberOfEntitiesToGiveFeedbackToSetting;
+            break;
+        }
+        return 0;
+    }
+
+    /**
+     * Get feedback participants who can see responses.
+     */
+    public List<FeedbackParticipantType> getShowResponsesTo() {
+        List<FeedbackParticipantType> showResponsesTo =
+                this.convertToFeedbackParticipantType(this.showResponsesTo);
+
+        // specially handling for contribution questions
+        // TODO: remove the hack
+        if (this.questionType == FeedbackQuestionType.CONTRIB
+                && this.giverType == FeedbackParticipantType.STUDENTS
+                && this.recipientType == FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF
+                && showResponsesTo.contains(FeedbackParticipantType.OWN_TEAM_MEMBERS)) {
+            // add the redundant participant type OWN_TEAM_MEMBERS even if it is just RECIPIENT_TEAM_MEMBERS
+            // contribution question keep the redundancy for legacy reason
+            showResponsesTo.add(FeedbackParticipantType.RECEIVER_TEAM_MEMBERS);
+        }
+
+        return showResponsesTo;
+    }
+
+    public List<FeedbackParticipantType> getShowGiverNameTo() {
+        return this.convertToFeedbackParticipantType(showGiverNameTo);
+    }
+
+    public List<FeedbackParticipantType> getShowRecipientNameTo() {
+        return this.convertToFeedbackParticipantType(showRecipientNameTo);
+    }
+
+    /**
+     * Converts a list of feedback visibility type to a list of feedback participant type.
+     */
+    private List<FeedbackParticipantType> convertToFeedbackParticipantType(
+            List<FeedbackVisibilityType> feedbackVisibilityTypes) {
+        return feedbackVisibilityTypes.stream().map(feedbackVisibilityType -> {
+            switch (feedbackVisibilityType) {
+            case STUDENTS:
+                return FeedbackParticipantType.STUDENTS;
+            case INSTRUCTORS:
+                return FeedbackParticipantType.INSTRUCTORS;
+            case RECIPIENT:
+                return FeedbackParticipantType.RECEIVER;
+            case GIVER_TEAM_MEMBERS:
+                return FeedbackParticipantType.OWN_TEAM_MEMBERS;
+            case RECIPIENT_TEAM_MEMBERS:
+                return FeedbackParticipantType.RECEIVER_TEAM_MEMBERS;
+            default:
+                assert false : "Unknown feedbackVisibilityType" + feedbackVisibilityType;
+                break;
+            }
+            return null;
```

Comment:
assert would cause the program to exit without catching. Perhaps you should do `log.severe` here?